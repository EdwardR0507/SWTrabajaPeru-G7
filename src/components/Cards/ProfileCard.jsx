/*Importamos las librerias principales*/
import React from "react";
import {
  Box,
  Card,
  CardHeader,
  Avatar,
  ListItem,
  ListItemAvatar,
  ListItemText,
  CardContent,
} from "@material-ui/core/";
import PhoneAndroidIcon from "@material-ui/icons/PhoneAndroid";
import MailIcon from "@material-ui/icons/Mail";
import RoomIcon from "@material-ui/icons/Room";
/*Declaramos la función principal*/
export default function ProfileCard(props) {
  /*Declaramos lo que nos va a retornar la funcion*/

  return props.user ? (
    <Card>
      <Box
        style={{
          textAlign: "center",
          display: "flex",
          justifyContent: "center",
          alignItems: "center",
        }}
      >
        <CardHeader
          avatar={
            <Avatar
              style={{ width: 150, height: 150, borderRadius: 75 }}
            ></Avatar>
          }
        />
      </Box>

      <CardHeader
        title={props.user.us_nombres}
        subheader="Especialista"
        style={{ textAlign: "center" }}
      />
      <CardContent>
        <ListItem>
          <ListItemAvatar>
            <Avatar>
              <PhoneAndroidIcon />
            </Avatar>
          </ListItemAvatar>
          <ListItemText
            primary="Teléfono / Celular"
            secondary={props.user.us_celular}
          />
        </ListItem>
        <ListItem>
          <ListItemAvatar>
            <Avatar>
              <MailIcon />
            </Avatar>
          </ListItemAvatar>
          <ListItemText
            primary="Correo electrónico"
            secondary={props.user.us_correo}
          />
        </ListItem>
        <ListItem>
          <ListItemAvatar>
            <Avatar>
              <RoomIcon />
            </Avatar>
          </ListItemAvatar>
          <ListItemText
            primary="Ubicación"
            secondary={`${props.user.us_departamento}/${props.user.us_provincia}/${props.user.us_distrito}`}
          />
        </ListItem>
      </CardContent>
    </Card>
  ) : (
    <div>Cargando...</div>
  );
}
