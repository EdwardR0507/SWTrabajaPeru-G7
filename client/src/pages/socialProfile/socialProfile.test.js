import SocialProfile from "./index";
import ProfileCard from "../../components/Cards/ProfileCard";
import React from "react";
import { shallow, mount, render } from "enzyme";
describe("<SocialProfile></SocialProfile>", () => {
  it('render `ProfileCard`', () => {
    const wrapper = shallow(<SocialProfile />);
    expect(wrapper).toMatchSnapshot()
  });
  it('should have an `ProfileCard` element', () => {
    const wrapper = shallow(<SocialProfile />);
    expect(wrapper.contains(<ProfileCard></ProfileCard>)).toBe(true);
  });
});
