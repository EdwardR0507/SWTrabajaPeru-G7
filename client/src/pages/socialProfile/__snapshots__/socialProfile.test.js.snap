// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SocialProfile></SocialProfile> render \`ProfileCard\` 1`] = `ShallowWrapper {}`;
